tools/FileTestDefs.sh

```diff
@@ -35,12 +35,42 @@ staged_grep() {
     git show ":./${@: -1}" | grep "${@:1:$(($#-1))}"
 }
 
-# Utility function for reporters that enables lots of decorators in grep
-# Works like staged_grep
+# Utility function for reporters that enables lots of decorators in
+# grep.  Works like staged_grep, except that it can take multiple file
+# arguments like real grep.  Accepts the same arguments as grep except
+# that any option taking an argument must use the --opt=arg form
+# instead of the short form.
 pretty_grep() {
-    echo -n -e "\033[0;35m${@: -1}\033[0m:"
-    git show ":./${@: -1}" | \
-        GREP_COLOR='1;37;41' grep -n $color_option "${@:1:$(($#-1))}"
+    local -a non_file_args
+    local file file_prefix
+    # This loop extracts all the flags and the pattern into
+    # non_file_args, leaving the filenames in $@.
+    while [ $# -gt 0 ] ; do
+        case "$1" in
+            --)
```

Comment:
What are two dashes in a grep command doing? I could not find it in the grep manual.

Reply:
Yeah, I was a bit surprised they did not document it explicitly.  It means to stop processing switches so you can pass other options (usually files, in this case a pattern) that begin with a `-` without them being interpreted as switches.  It is a standard GNU-style option parsing thing, so I guess since they didn't implement it they left it out (intentionally or accidentally).

Reply:
Would it be maybe be good to add a comment on what it does if it's not in the grep manual?

Reply:
Ah thanks, I remember now where I have used this before. 

@wthrowe However, your implementation does not work when `--` is used after the pattern.

Reply:
I didn't think of that.  Handling that case sounds annoying.  Would you be OK with a comment telling people not do to that instead?  (Any "--" case people want can be trivially rewritten into the case that is already handled.)